jenkinsbuilder/indexdindexymltransform.py

```diff
@@ -0,0 +1,196 @@
+from yaml import load, safe_dump
+from glob import glob
+import os
+from argparse import ArgumentParser
+
+
+class IndexGenerator:
+    """Thus class transforms index.d to index.yml and vice versa, Genearting one or the other, as needed"""
+
+    def __init__(self, indexd_path="./index.d", index_yml="./index.yml", reverse=False):
+        """Initialze a generator, giving path of index.d and index.yml and specifying is reverse transform needs to be
+        done"""
+
+        self._index_dotd = indexd_path
+        self._index_yml = index_yml
+        self._reverse = reverse
+
+        # If not reverse, check index.d folder for existence.
+        if not self._reverse:
+                if not os.path.exists(self._index_dotd):
+                    raise ValueError("Invalid path specified.")
+
+        # Otherwise, check for index.yml existance
+        else:
+            if not os.path.exists(self._index_yml):
+                raise ValueError("Invalid path specified")
+
+        return
+
+    def _genindexyml(self, potential_index_files):
+        """Generate index.yml from index.d files"""
+
+        current_index = 1
```

Comment:
There's no need to track index using ``current_index`` var, you can just use ``enumerate`` in line 39, and use ``n`` as as the index, unless you have a special reason to do so.